### ISSUE TITLE
kola: `iso-offline-install (uefi + metal4k)` fails on f37 and f38 with latest grub2

### ISSUE THREAD

**Issue by jlebon** [coreos/fedora-coreos-tracker#1352]
```
10:39:23  FAIL: iso-offline-install (uefi + metal4k) (10m0s)
10:39:23      timed out after 10m0s
10:39:23  Error: timed out after 10m0s
10:39:23  2022-11-24T15:39:14Z cli: timed out after 10m0s
10:39:23  error: failed to execute cmd-kola: exit status 1
```

Looking at the console, we just have:

```
[2J[01;01H[=3h[2J[01;01H[2J[01;01H[=3h[2J[01;01HBdsDxe: failed to load Boot0002 "UEFI Misc Device" from PciRoot(0x0)/Pci(0x3,0x0): Not Found
BdsDxe: loading Boot0001 "UEFI QEMU DVD-ROM QM00011 " from PciRoot(0x0)/Pci(0x1F,0x2)/Sata(0x5,0xFFFF,0x0)
BdsDxe: starting Boot0001 "UEFI QEMU DVD-ROM QM00011 " from PciRoot(0x0)/Pci(0x1F,0x2)/Sata(0x5,0xFFFF,0x0)
[02;32HGRUB version 2.06


   Minimal BASH-like line editing is supported. For the first word, TAB   

   lists possible command completions. Anywhere else TAB lists possible   

   device or file completions.                                            



grub> 
```

[iso-offline-install.zip](https://github.com/coreos/fedora-coreos-tracker/files/10086052/iso-offline-install.zip)

**Comment by jlebon** [comment]
```
07:46:11  Upgraded:
07:46:11    container-selinux 2:2.191.0-1.fc38 -> 2:2.193.0-1.fc38
07:46:11    grub2-common 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    grub2-efi-x64 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    grub2-pc 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    grub2-pc-modules 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    grub2-tools 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    grub2-tools-minimal 1:2.06-64.fc38 -> 1:2.06-67.fc38
07:46:11    irqbalance 2:1.9.0-1.fc37 -> 2:1.9.1-1.fc38
07:46:11    libbsd 0.10.0-10.fc37 -> 0.11.7-1.fc38
07:46:11    libgcc 12.2.1-3.fc38 -> 12.2.1-4.fc38
07:46:11    libselinux 3.4-5.fc37 -> 3.4-6.fc38
07:46:11    libselinux-utils 3.4-5.fc37 -> 3.4-6.fc38
07:46:11    libsemanage 3.4-5.fc37 -> 3.4-6.fc38
07:46:11    libsepol 3.4-3.fc37 -> 3.4-4.fc38
07:46:11    libstdc++ 12.2.1-3.fc38 -> 12.2.1-4.fc38
07:46:11    mpfr 4.1.1-1.fc38 -> 4.1.1-2.fc38
07:46:11    ostree 2022.6-1.fc38 -> 2022.7-2.fc38
07:46:11    ostree-libs 2022.6-1.fc38 -> 2022.7-2.fc38
07:46:11    policycoreutils 3.4-6.fc37 -> 3.4-7.1.fc38
07:46:11    selinux-policy 37.14-1.fc38 -> 38.1-1.fc38
07:46:11    selinux-policy-targeted 37.14-1.fc38 -> 38.1-1.fc38
07:46:11    shadow-utils 2:4.13-2.fc38 -> 2:4.13-3.fc38
07:46:11    shadow-utils-subid 2:4.13-2.fc38 -> 2:4.13-3.fc38
07:46:11  Added:
07:46:11    libmd-1.0.4-2.fc38.x86_64
```

Possibly `grub2-efi-x64`? Notable changes:
- https://src.fedoraproject.org/rpms/grub2/c/1af394246eccd2dbc1fdbe3f199ba28b0e36b935?branch=rawhide
- https://src.fedoraproject.org/rpms/grub2/c/0ccadff7a2be9fcc858e0a1cb7825beb5d81f8c6?branch=rawhide

/cc @frozencemetery @AdamWill

**Comment by AdamWill** [comment]
Can you isolate which build of grub2 between 64 and 67 breaks it? Assuming grub is the cause, which seems reasonably likely...

**Comment by jlebon** [comment]
First build to fail is 66. So that points to https://src.fedoraproject.org/rpms/grub2/c/0ccadff7a2be9fcc858e0a1cb7825beb5d81f8c6?branch=rawhide. @frozencemetery, let me know if you'd like me to file a BZ for this or if you have any tips to debug this further.

**Comment by jlebon** [comment]
Seems like we're hitting this in f37 too now. Which makes sense since that same commit is on the f37 branch: https://src.fedoraproject.org/rpms/grub2/c/0ccadff7a2be9fcc858e0a1cb7825beb5d81f8c6?branch=f37

Pin in https://github.com/coreos/fedora-coreos-config/pull/2102

**Comment by frozencemetery** [comment]
The only thing I can think of is that you might be hitting space constraints - grub*.efi grows by about a megabyte with these changes.  Is that possible?

**Comment by jlebon** [comment]
Hmm, latest f37 GRUB2 is also breaking our PXE + UEFI tests: https://github.com/coreos/coreos-assembler/pull/3250.

> The only thing I can think of is that you might be hitting space constraints - grub*.efi grows by about a megabyte with these changes. Is that possible?

Hmm, it should be fine though I'm not 100% sure which space we're talking about. Do you mean the EFI partition or RAM or something else?

**Comment by frozencemetery** [comment]
Well, that depends on what you're tying to do.  I see mention of PXE in the linked issue - is this about network boot or local boot?

**Comment by frozencemetery** [comment]
It would also help my own sanity if it could be checked that the contents of grub2-efi-x64 match what''s ending up in the ESP here.

**Comment by jlebon** [comment]
> Well, that depends on what you're tying to do. I see mention of PXE in the linked issue - is this issue using pxe or local boot?
> 
> I had assumed local and was asking about ESP.

Right sorry, this is confusing and assumed some knowledge of our systems. Let me summarize and clarify. There are two failures right now:
1. Booting the FCOS live ISO in UEFI mode with the latest f37/f38 GRUB2 fails with the console output in https://github.com/coreos/fedora-coreos-tracker/issues/1352#issue-1463560892.
2. Booting FCOS via a PXE server with the latest f37 `grubx64.efi` fails with the console output in https://github.com/coreos/coreos-assembler/pull/3250#issue-1470105620.

So in neither case is an ESP involved AFAIK. In both cases, the machines have 4G of RAM.

I've uploaded an f38 ISO [here](https://jlebon.fedorapeople.org/fedora-coreos-38.20221124.dev.2-live.x86_64.iso). You should be able to reproduce it locally. Steps I used:

```sh
$ cp /usr/share/edk2/ovmf/OVMF_VARS.fd vars.fd
$ qemu-system-x86_64 -m 4G -machine q35 \
  -drive file=/usr/share/edk2/ovmf/OVMF_CODE.fd,if=pflash,format=raw,unit=0,readonly=on,auto-read-only=off \
  -drive file=$PWD/vars.fd,if=pflash,format=raw,unit=1,readonly=off,auto-read-only=off \
  -cdrom fedora-coreos-38.20221124.dev.2-live.x86_64.iso
```

Would be happy to meet also and do a screenshare so we can try to debug this together.

**Comment by dustymabe** [comment]
Note that in https://github.com/coreos/fedora-coreos-config/pull/2101 and https://github.com/coreos/fedora-coreos-config/pull/2102 we pinned `grub2` for `aarch64` and `x86_64`. We should have also pinned for `ppc64le` since there are `grub2` packages in that arch too.

**Comment by frozencemetery** [comment]
This may be fixed in grub2-2.06-72.fc38.

**Comment by jlebon** [comment]
> This may be fixed in grub2-2.06-72

Thanks! I can confirm both issues in https://github.com/coreos/fedora-coreos-tracker/issues/1352#issuecomment-1332806646 are fixed.

**Comment by dustymabe** [comment]
We've dropped the pins in:

- https://github.com/coreos/coreos-assembler/pull/3306
- https://github.com/coreos/fedora-coreos-config/pull/2152
- https://github.com/coreos/fedora-coreos-config/pull/2153
